cartographer/mapping_2d/local_trajectory_builder.cc

```diff
@@ -73,8 +79,19 @@ const Submaps* LocalTrajectoryBuilder::submaps() const { return &submaps_; }
 sensor::LaserFan LocalTrajectoryBuilder::BuildCroppedLaserFan(
     const transform::Rigid3f& tracking_to_tracking_2d,
     const sensor::LaserFan& laser_fan) const {
+  LaserFan fan{laser_fan.origin, {}, {}, laser_fan.reflectivities};
+  for (const Eigen::Vector3f& return_ : laser_fan.returns) {
+    const float range = (return_ - laser_fan.origin).norm();
+    if (options_.laser_min_range() <= range &&
+        range <= options_.laser_max_range()) {
+      fan.returns.push_back(return_);
+    }
+  }
+  for (const Eigen::Vector3f& miss : laser_fan.misses) {
+    fan.misses.push_back(options_.laser_missing_echo_ray_length);
```

Comment:
This seems strange: Adding a scalar value to the point cloud of misses? "options_.laser_missing_echo_ray_scale * miss"?